aspnet/mvc/views/working-with-forms.rst

```diff
@@ -0,0 +1,621 @@
+
+Syntax:
+
+.. code-block:: HTML
+
+  <input asp-for="<Expression Name>" /> 
+
+The Input Tag Helper:
+
+- Generates the ``id`` and ``name`` HTML attributes for the expression name specified in the ``asp-for`` attribute.  ``asp-for="Property1.Property2"`` is equivalent to ``m => m.Property1.Property2``, that is the attribute value literally is part of an expression. The name of the expression is what's used for the ``asp-for`` attribute value.
+- Sets the HTML ``type`` attribute value based on the model type and  `data annotation <https://msdn.microsoft.com/en-us/library/system.componentmodel.dataannotations.aspx>`__ attributes applied to the model property
+- Generates `HTML5 <https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/HTML5>`__  validation attributes from `data annotation <https://msdn.microsoft.com/en-us/library/system.componentmodel.dataannotations.aspx>`__ attributes applied to model properties
+- Will not overwrite the HTML ``type`` attribute value when one is specified 
+- Has an HTML Helper feature overlap with ``Html.TextBoxFor()`` and ``Html.EditorFor()``. See the **HTML Helper alternative to Input Tag Helper** section for details.
+- Provides strong typing. If the name of the property changes and you don't update the Tag Helper you'll get an error similar to the following:
+
+.. code-block:: HTML
+
+     An error occurred during the compilation of a resource required to process
+     this request. Please review the following specific error details and modify
+     your source code appropriately.
+    
+     Type expected
+      'RegisterViewModel' does not contain a definition for 'Email' and no
+      extension method 'Email' accepting a first argument of type 'RegisterViewModel'
+      could be found (are you missing a using directive or an assembly reference?)
+
+The ``Input`` Tag Helper sets the HTML ``type`` attribute based on the .NET type. The following table lists some common .NET types and generated HTML type (not every .NET type is listed). 
+
++---------------------+--------------------+
+|.NET type            |  Input Type        |  
++=====================+====================+
+|Bool                 |  type="checkbox"   |
++---------------------+--------------------+  
+|String               |  type="text"       |
++---------------------+--------------------+  
+|DateTime             |  type="datetime"   |
++---------------------+--------------------+  
+|Byte                 |  type="number"     |
++---------------------+--------------------+  
+|Int                  |  type="number"     |
++---------------------+--------------------+  
+|Single, Double       |  type="number"     |
++---------------------+--------------------+  
+
+The following table shows some common`data annotations <https://msdn.microsoft.com/en-us/library/system.componentmodel.dataannotations.aspx>`__ attributes that the input tag helper will map to specific input types (not every validation attribute is listed):
```

Comment:
Something is off with the syntax in this paragraph. Looks a mess here and in the HTML output.